src/js/pages/GranthIndex/components/GranthIndex.tsx

```diff
@@ -1,38 +1,150 @@
+  const changeSelection = useCallback(
+    (path: string[]) => {
+      if (path.join() == selection.join()) {
+        path.pop();
+      }
+      setSelection(path);
+    },
+    [selection]
+  );
+
   useEffect(() => {
     pageView('/index');
     // making sure the element is rendered on DOM before we get to that
     setTimeout(() => scrollToHash(hash), 800);
-  }, [])
+  }, []);
+
+  // Recursive function that keeps churning lists until no subSets are left
+  // path is a list that helps keep the parents open when a child is clicked upon
+  const renderList = useCallback(
+    (
+      subSets: IRaagIndex[],
+      source: IRaagIndex['source'] = 'G',
+      path: string[] = []
+    ) => {
+      return subSets.map(
+        (
+          {
+            name,
+            pages: [from, to],
+            source: localSource,
+            description,
+            subSets,
+            // Give a default value if highlight is undefined
+            highlight = 0,
+          },
+          i
+        ) => {
+          if (localSource == null) {
+            localSource = source;
+          } else {
+            source = localSource;
+          }
+
+          return (
+            <div
+              key={name}
+              style={{ marginLeft: '40px' }}
+              id={sanitizeHash(...path, name)}
+            >
+              <li
+                className="list--item"
+                style={{
+                  animationDelay: i < 15 ? `${20 * i}ms` : '0',
+                }}
+                onClick={() => {
+                  changeSelection(path.concat([name]));
+                }}
+              >
+                <div className="index-list">
+                  <div className="index-list--header">
+                    <p>{name}</p>
+                    <p>
+                      <Link
+                        className="index-list--url"
+                        to={toAngURL({
+                          ang: from,
+                          source: localSource,
+                          highlight,
+                        })}
+                      >
+                        {from}
+                      </Link>{' '}
+                      to{' '}
+                      <Link
+                        className="index-list--url"
+                        to={toAngURL({
+                          ang: to,
+                          source: localSource,
+                          highlight,
+                        })}
+                      >
+                        {to}
+                      </Link>
+                    </p>
+                  </div>
+                  {isSelected(name) && (
+                    <div className="index-list--description">{description}</div>
+                  )}
+                </div>
+              </li>
+              {isSelected(name) && subSets && subSets.length > 0
+                ? renderList(subSets, localSource, path.concat([name]))
+                : null}
+            </div>
```

Comment:
@yudhv veerji, here we are opening the sublist from the main list item [if we have clicked on it], right?

Reply:
yes